Code/Framework/AzCore/AzCore/Asset/AssetCommon.h

```diff
@@ -1203,7 +1203,14 @@ namespace AZ
         {
             if (m_assetData && m_assetData->GetId().IsValid())
             {
-                return AssetInternal::ReloadAsset(m_assetData, m_loadBehavior);
+                auto asset = AssetInternal::ReloadAsset(m_assetData, m_loadBehavior);
+
+                // Only assign on success so we don't lose the assetId/assetType if the above failed
+                if (asset)
```

Comment:
if this fails, is it still valid to have the *this pointing to a failed reloaded asset?